### PR TITLE
Fixing MISRA violation for FreeRTOS_IP*, FreeRTOS_DHCP*

### PR DESCRIPTION
**Pull request by xuelix**
Description
-----------

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_DHCP.c [hunk 1]
```diff
@@ -312,8 +321,12 @@
                 }
             }
         }
+        else
+        {
+            /* do nothing, coverity happy */
+        }
 
-        if( ( pxEndPoint != NULL ) && ( xDoProcess != pdFALSE ) )
+        if( xDoProcess != pdFALSE )
```
Comment (moninom1) [inline source/FreeRTOS_DHCP.c:329]: What is the reason for removing pxEndPoint NULL check?

File: source/FreeRTOS_IP.c [hunk 2]
```diff
@@ -190,6 +178,10 @@ uint16_t usPacketIdentifier = 0U;
  * reference. */
 const MACAddress_t xBroadcastMACAddress = { { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff } };
 
+/* __XX__ temporary for coverity build */
+/** @brief Structure that stores the netmask, gateway address and DNS server addresses. */
+NetworkAddressingParameters_t xNetworkAddressing = { 0, 0, 0, 0, 0 };
+
```
Comment (moninom1) [inline source/FreeRTOS_IP.c:184]: Why is this needed ?

Comment (htibosch) [reply source/FreeRTOS_IP.c]: Do you mean why it is initialised, or what the function is of `xNetworkAddressing`?
It is initialised just to show it explicitly.
The struct will be dropped as soon as we have multiple interfaces. The struct will be replaced by end-points.

Comment (moninom1) [reply source/FreeRTOS_IP.c]: Yes hein, as we have moved to end point based implementation , i removed it in previous PR, was just checking why was it needed again for fixing coverity.

**Review by moninom1 — APPROVED** [review]
